unity-renderer/Assets/Scripts/MainScripts/DCL/WorldRuntime/KernelCommunication/RPC/GeneratedCode/RendererProtocolService.gen.cs

```diff
@@ -0,0 +1,44 @@
+// AUTOGENERATED, DO NOT EDIT
+// Type definitions for server implementations of ports.
+// package: 
+// file: RendererProtocol.proto
+using System.Collections.Generic;
+using Cysharp.Threading.Tasks;
+using Google.Protobuf;
+using rpc_csharp.protocol;
+using rpc_csharp;
+
+public abstract class CRDTService<Context>
+{
+    public const string ServiceName = "CRDTService";
+
+    public delegate UniTask<CRDTResponse> SendCrdt(CRDTManyMessages request, Context context);
+
+    public delegate IEnumerator<CRDTManyMessages> CrdtNotificationStream(CRDTStreamRequest request, Context context);
```

Comment:
Why some are tasks and other corroutines?

Reply:
We had lot of back and forth thinking about that.
The idea why procedure calls (`SendCrdt`) are tasks is because maybe in some scenarios we want to process something "async" there, which is almost never, but we didn't feel comfortable to remove that possibility.
And streams (`CrdtNotificationStream`) are not actually running on a coroutine, it's just and old school enumerator (using iterator.MoveNext), we didn't think it was reasonable to enforce the extra work of every implementation using tasks when it is a "almost never" scenario and can have an easy workaround like `while(task_not_completed) yield return null`